solar_crypto/transactions/builder/vote.py

```diff
@@ -1,38 +1,96 @@
+import re
 import typing
+from functools import cmp_to_key
+from math import trunc
 
-from solar_crypto.constants import TRANSACTION_VOTE
-from solar_crypto.identity.address import address_from_passphrase
+from solar_crypto.constants import SOLAR_TRANSACTION_VOTE, TRANSACTION_TYPE_GROUP
 from solar_crypto.transactions.builder.base import BaseTransactionBuilder
 
 
 class Vote(BaseTransactionBuilder):
 
-    transaction_type = TRANSACTION_VOTE
+    transaction_type = SOLAR_TRANSACTION_VOTE
+    typeGroup = TRANSACTION_TYPE_GROUP.SOLAR.value
 
-    def __init__(self, vote=None, fee=None):
-        """Create a second signature registration transaction
-
-        Args:
-            vote (str): address of a delegate you want to vote
-            fee (int, optional): fee used for the transaction (default is already set)
-        """
+    def __init__(self):
         super().__init__()
 
         self.transaction.asset["votes"] = []
-        if vote:
-            self.transaction.asset["votes"].append(vote)
 
-        if fee:
-            self.transaction.fee = fee
-
-    def set_votes(self, votes: typing.List[str]):
-        """Set votes/unvotes
+    def set_votes(
+        self,
+        votes: typing.Union[typing.List[str], typing.Dict[str, typing.Union[int, float]]] = dict,
+    ):
+        """Set votes
 
         Args:
-            votes (List[str]): list of votes
+            votes
         """
+        vote_object: typing.Dict[str, typing.Union[float, int]] = {}
+
+        if isinstance(votes, list):
+            vote_list = filter(lambda vote: not vote.startswith("-"), votes)
+            vote_list = list(map(lambda vote: vote[1:], vote_list))
+
+            if len(vote_list) == 0:
+                self.transaction.asset["votes"] = {}
+                return
+
+            weight = round((trunc((100 / len(vote_list)) * 100) / 100) * 100)
```

Comment:
per @alessiodf 
round isn't necessary and was removed a few days ago .

The current Ts representation is:
`const weight: number = Math.trunc((((100 / voteArray.length) * 100) / 100) * 100);`

https://github.com/Solar-network/core/blob/develop/packages/crypto/src/transactions/builders/transactions/solar/vote.ts#L25